icicle-cpu/src/lifter/mod.rs

```diff
@@ -678,7 +699,7 @@ impl BlockLifter {
                     next_addr
                 }
                 BlockResult::Exit(addr) => break Target::External(addr.into()),
-                BlockResult::Invalid => break Target::Invalid,
+                BlockResult::Invalid(e) => return Err(e),
```

Comment:
We can't return here directly because there might be previous instructions in the block before the invalid instruction that we still want to be able to execute them.

The strategy I was using here was to was basically to allow blocks that end in an invalid instruction to still be generated then handle actually raising it to an exception when the end of the block is reached.

Reply:
I think what should be done here is that the decode error should be injected as `pcode::Op::Exception` to allow things to work better.

(Ideally also handling the optimization when there are not extra instructions in the block (like below))

Reply:
Thanks for the feedback! This makes sense and I'll create a test case for it and see if I can get it to work. This also explains my confusion with the `BlockStartEndEqual`, this would have to be replaced with the appropriate `DecodeError`.